src/main/groovy/de/iteratec/osm/result/dao/query/transformer/MeasurandPercentileDataTransformer.groovy

```diff
@@ -0,0 +1,68 @@
+package de.iteratec.osm.result.dao.query.transformer
+
+import de.iteratec.osm.result.SelectedMeasurand
+import de.iteratec.osm.result.dao.EventResultProjection
+import de.iteratec.osm.result.dao.query.AggregationUtil
+import de.iteratec.osm.result.dao.query.ProjectionProperty
+
+class MeasurandPercentileDataTransformer implements EventResultTransformer {
+    Set<ProjectionProperty> baseProjections
+    List<SelectedMeasurand> selectedMeasurands
+    private int percentage
+
+    MeasurandPercentileDataTransformer(int percentage, Set<ProjectionProperty> baseProjections, List<SelectedMeasurand> selectedMeasurands) {
+        this.percentage = percentage
+        this.baseProjections = baseProjections
+        this.selectedMeasurands = selectedMeasurands
+    }
+
+    @Override
+    List<EventResultProjection> transformRawQueryResult(List<Map> rawQueryData) {
+        List<String> measurandNames = selectedMeasurands.findAll { !it.selectedType.isUserTiming() }.collect {
+            it.databaseRelevantName
+        }
+        Map<String, List<Map>> groupedRawData = groupRawDataByAggregators(rawQueryData, baseProjections)
+        Set<EventResultProjection> groupedAndSummarized = summarizeGroupedRawData(groupedRawData, baseProjections, measurandNames)
+        return getPercentileFromSummarizedData(groupedAndSummarized, measurandNames) as List
+    }
+
+    private Map<String, List<Map>> groupRawDataByAggregators(List<Map> rawData, Set<ProjectionProperty> baseProjections) {
+        Map<String, List<Map>> groupedRawData = [:].withDefault { [] }
+        rawData.each { ungrouped ->
+            String key = AggregationUtil.generateGroupedKeyForAggregations(ungrouped, baseProjections)
+            groupedRawData.get(key) << ungrouped
+        }
+        return groupedRawData
+    }
+
+    private Set<EventResultProjection> summarizeGroupedRawData(Map<String, List<Map>> groupedRawData, Set<ProjectionProperty> baseProjections, List<String> measurandNames) {
+        Map<EventResultProjection, List<Map>> justAHelperMap = [:]
```

Comment:
i am uncertain if justAHelperMap is a good name, any better name to describe its purpose?